src/pyqasm/exceptions.py

```diff
@@ -18,10 +18,21 @@
 """
 
 import logging
+import os
+import sys
 from typing import Optional, Type
 
-from openqasm3.ast import Span
+from openqasm3.ast import QASMNode, Span
 from openqasm3.parser import QASM3ParsingError
+from openqasm3.printer import dumps
+
+# Define a custom logger for the module
+handler = logging.StreamHandler()
+handler.setFormatter(logging.Formatter("%(levelname)s:%(name)s: %(message)s"))
+
+logger = logging.getLogger("pyqasm")
+logger.addHandler(handler)
+logger.setLevel(logging.WARNING)
```

Comment:
We want to be careful that doing this doesn't cause the user to not be able to modify the logging cofigs themselves. For example, if a user wanted to later on set the log level to something different, they should still be able to do that. See https://github.com/qBraid/qBraid/issues/941 as an example where we had a similar logging config hard-coded in the SDK which caused an inconvenience to an end-user.